modules/commands/os_dns.cpp

```diff
@@ -349,11 +349,13 @@ class CommandOSDNS : public Command
 				DNSZone *z = DNSZone::Find(zone);
 				if (!z)
 				{
+					delete z;
 					source.Reply(_("Zone %s does not exist."), zone.c_str());
 					return;
 				}
 				else if (z->servers.count(s->GetName()))
 				{
+					delete s;
```

Comment:
this is wrong

Reply:
@Adam-, Both delete statements?

Reply:
@Adam-, fix it.
https://github.com/anope/anope/pull/295/commits/4fb0a3908ddd4305f06db76d8e60330a92e4a28e